src/app/layout.tsx

```diff
@@ -46,6 +65,12 @@ export default async function RootLayout({
 }: {
   children: ReactNode;
 }) {
+  // @ts-ignore FIXME testing server-side metrics
+  const pathname = children?.props?.childProp?.segment;
```

Comment:
@flozia / @Vinnl - is this the best way to get the pathname from a server-side component? Note that it shows something different when using the dev server (e.g. `(nextjs_migration)`), it was the pathname with a real HTTP server when I tested it.

Reply:
I'd like a more reliable way to capture things like impressions and clicks, and the server is a pretty reliable way to do that. We already have HTTP access logs that are in a structured format from nginx that we could use here if it doesn't make sense to use NextJS for it, that might be a better way to go.

Reply:
I'd like to just move as much to first-party and server-side collection as possible, and for things where client events are useful (and the user isn't blocking or opting out) we can send those and join on them.

There is a line somewhere between "server-side logs" and "tracking user behavior that must be deleted if users request" which is something on my mind as well. Somewhat ironically, in order to support deletion requests, we need to be able to associate data with a particular user, which may be something we need NextJS to log in a structured way that gets stored such that we can act on things like deletion requests.

We don't really need to be logging this here though, I'll go ahead and remove it and we can tackle impressions and clicks separately, that really deserves its own discussion.